packages/explat-client-react-helpers/src/test/index.tsx

```diff
@@ -150,3 +150,38 @@ describe( 'Experiment', () => {
 		expect( container.textContent ).toBe( 'default-2' );
 	} );
 } );
+
+describe( 'ProvideExperimentData', () => {
+	it( 'should correctly provide data', async () => {
+		const exPlatClient = createMockExPlatClient();
+		const { ProvideExperimentData } = createExPlatClientReactHelpers( exPlatClient );
+
+		const controllablePromise1 = createControllablePromise< ExperimentAssignment >();
+		( exPlatClient.loadExperimentAssignment as jest.MockedFunction<
+			typeof exPlatClient.loadExperimentAssignment
+		> ).mockImplementationOnce( () => controllablePromise1.promise );
+
+		const capture = jest.fn();
+		render(
+			<ProvideExperimentData name="experiment_a">
+				{ ( isLoading, experimentAssignment ) => (
+					<>{ capture( isLoading, experimentAssignment ) }</>
+				) }
+			</ProvideExperimentData>
```

Comment:
I've never seen this before! Clever!

Reply:
lol I haven't seen it either, not sure if it is good practice 🤞